tuning/sft_trainer.py

```diff
@@ -340,32 +340,36 @@ def train(
     # this validation, we just drop the things that aren't part of the SFT Config and build one
     # from our object directly. In the future, we should consider renaming this class and / or
     # not adding things that are not directly used by the trainer instance to it.
+
     transformer_train_arg_fields = [x.name for x in dataclasses.fields(SFTConfig)]
     transformer_kwargs = {
         k: v
         for k, v in train_args.to_dict().items()
         if k in transformer_train_arg_fields
     }
+
     training_args = SFTConfig(**transformer_kwargs)
+    training_args.dataset_text_field = (
+        dataset_text_field  # Inject the dataset text field into the training args
+    )
+    training_args.packing = packing  # Inject packing into the training args
+    training_args.max_seq_length = (
+        max_seq_length  # Specify max sequence length for the dataset
+    )
```

Comment:
```suggestion
    additional_args = {
        "dataset_text_field": dataset_text_field,
        "packing": packing,
        "max_seq_length": max_seq_length,
    }
    if is_pretokenized_dataset(
        data_args.training_data_path or data_args.validation_data_path
    ):
        additional_args["dataset_kwargs"] = {"skip_prepare_dataset": True}
    training_args = SFTConfig(**transformer_kwargs, **additional_args)
```

This would be a neater alternative I believe and expanding the additional_args dict would be easy for future new args so the code is less scattered. WDYT? Thanks

Reply:
This looks good to me. I agree, this way would be easier to add more args in the future. I will make changes locally and test to see if everything is working fine, then commit suggestion.

Reply:
I'm getting an error: 
`trl.trainer.sft_config.SFTConfig() got multiple values for keyword argument 'packing'`

Seems like `packing` is already a key from train_args dict, and it can't be duplicated by additional args. Seems like previously the `packing` from `training_args` was being overwritten on line 355?

Reply:
@Luka-D 
Can you update this PR with your current code what you have, so that I can see where its going wrong?

Reply:
@kmehant I've made a commit just now with my changes. Commit https://github.com/foundation-model-stack/fms-hf-tuning/pull/399/commits/a0b736041478b6bd488423b5465d67e611045cc9

Reply:
Can we possibly merge `transformer_kwargs` and `additional_args` dictionaries before invoking `SFTConfig()`?

Reply:
Thank you @Luka-D

Reply:
I think lets clean up the code a bit, lets avoid usage of variables packing and max_seq_len rather we can use it directly from train_args. 

For instance, we no need to even introduce `packing` variable here - https://github.com/foundation-model-stack/fms-hf-tuning/pull/399/files#diff-7d8f6f8626340337d03a2f04fec2840d2ede32b0c7af2640eed9759af45bb3cfR308 and use train_args.packing everywhere one example - https://github.com/foundation-model-stack/fms-hf-tuning/pull/399/files#diff-7d8f6f8626340337d03a2f04fec2840d2ede32b0c7af2640eed9759af45bb3cfR314

In that case additional_args would only have dataset_text_field arg and rest all (packing, max_seq_len) would fall in place automatically at this part of the code - https://github.com/foundation-model-stack/fms-hf-tuning/pull/399/files#diff-7d8f6f8626340337d03a2f04fec2840d2ede32b0c7af2640eed9759af45bb3cfR344-R349.


I am thinking of this approach, this would also clean up this code a bit, WDYT?

Reply:
@kmehant Sounds good, I'll remove `packing` and `max_seq_len`, keep `dataset_test_field` and `dataset_kwargs` in `additional_args`. I'll make the changes and commit

Reply:
@kmehant I removed `packing` and `max_seq_length` in commit https://github.com/foundation-model-stack/fms-hf-tuning/pull/399/commits/830f8c8513e69e25dc0dcae5b8d861b9da7dbbc9. `["dataset_kwargs"] = {"skip_prepare_dataset": True}` is still there on line 357, assuming that the conditional above is satisfied. Ran tuning locally and ran tox with no issues. Let me know your thoughts.